wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -555,18 +555,28 @@ public WorkspaceDto updateEnvironment(@ApiParam("The workspace id")
                                           @PathParam("name")
                                           String envName,
                                           @ApiParam(value = "The environment update", required = true)
-                                          EnvironmentDto update) throws ServerException,
-                                                                        BadRequestException,
-                                                                        NotFoundException,
-                                                                        ConflictException,
-                                                                        ForbiddenException {
+                                          EnvironmentDto update,
+                                          @ApiParam("New name of the environment")
+                                          @QueryParam("new-name")
```

Comment:
Can you please explain why do we need this qp, it seems that everything should be fine with incoming environment